### ISSUE TITLE
Inconsistent behavior of renderTarget between r70 & r73

### ISSUE THREAD

**Issue by jyunming-chen** [mrdoob/three.js#8858]
##### Description of the problem 
I tried to use renderTarget to perform multi-pass rendering
(in particular, render a scene to an FBO, use shader to make it monochrome,
then another pass to add colored focus object, blah, blah, blah)

Anyway, in one of the passes,  I need to set up the depth buffer (while disabling the color buffer output).
Therefore, I have to set the material (colorWrite) to false. This seems to work only in r73.

But the result of my previous passes failed to show up.
But if I change to r70, the render-to-texture is fine again.
...
jsfiddle: http://jsfiddle.net/jmcjc5u/vuutsvsc/2/

##### Three.js version

- [ ] Dev
- [ ] r76
- [ ] ...

##### Browser

- [] All of them
- [ x] Chrome
- [ ] Firefox
- [ ] Internet Explorer

##### OS

- [ ] All of them
- [x ] Windows
- [ ] Linux
- [ ] Android
- [ ] IOS

**Comment by mrdoob** [comment]
Getting a Error 404 on that link.

**Comment by jyunming-chen** [comment]
Thank you so much for replying. and sorry for an immature bug report.

I want to render a mostly monochrome scene with a colored focus, like this
photo:
[image: Inline image 1]
Anyways, here's what I got finally: https://jsfiddle.net/jmcjc5u/rf2e6qrt/

first pass: render scene (without the focus "box") to renderTarget,
second pass: render a plane in orthographic camera with the rtTexture, then
make it monochrome using shader

third pass: render the scene again, with colorWrite false, to set up the
"blocker" in depth buffer
final pass: render the focus red "box"

What I found: colorWrite false: was not working in r70 (still writes to
colorBuffer)

As for the bug I reported, somehow, it is alright now.




On Mon, May 16, 2016 at 11:55 PM, Mr.doob <notifications@github.com> wrote:

> Getting a Error 404 on that link.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/mrdoob/three.js/issues/8858#issuecomment-219463225>
>



-- 
Jyun-Ming Chen (陳俊銘)
Computer Graphics Laboratory
Dept. of Computer Science and Engineering
Tatung University
Taipei, Taiwan
